File: examples/robot/RobotExample.java
package org.sdk.lebai.robot;
import org.sdk.lebai.Robot;

public final class RobotExample {
  public static void main(String[] args) {
    if (args.length < 1) {
      System.err.println("Usage: RobotExample <robot_ip> [sim] [--move]");
      return;
    }

    boolean simulator = false;
    boolean runMotion = false;
    for (int i = 1; i < args.length; ++i) {
      if ("sim".equals(args[i])) {
        simulator = true;
      } else if ("--move".equals(args[i])) {
        runMotion = true;
      }
    }

    System.out.println("Example java robot sdk...");
    try (Robot robot = new Robot(args[0], simulator)) {
      System.out.println("System: " + robot.getNative().get_system_info().getName());
      System.out.println("State: " + robot.getNative().get_robot_state());
      System.out.println("Estop reason: " + robot.getNative().get_estop_reason());

      if (!runMotion) {
        System.out.println("Motion skipped. Pass --move to run the motion demo.");
        return;
      }

      robot.startSys();
      robot.moveJ(new double[] {0.0, 0.0, 0.0, 0.0, 0.0, 0.5}, 0.5, 0.5, 0.0, 0.0);
      robot.moveJ(new double[] {0.0, 0.0, 0.0, 0.0, 0.0, 0.0}, 0.5, 0.5, 0.0, 0.0);
      robot.waitMove(0);
    }
  }

  private RobotExample() {}
}

File: examples/robot/RobotExample.cs
using System;
using lebai;

namespace lebai.app
{
    class RobotExample
    {
        static int Main(string[] args)
        {
            if (args.Length < 1)
            {
                Console.Error.WriteLine($"Usage: {AppDomain.CurrentDomain.FriendlyName} <robot_ip> [sim] [--move]");
                return 1;
            }

            bool simulator = Array.IndexOf(args, "sim") >= 0;
            bool runMotion = Array.IndexOf(args, "--move") >= 0;

            using (var robot = new Robot(args[0], simulator))
            {
                Console.WriteLine($"System: {robot.Native.get_system_info().name}");
                Console.WriteLine($"State: {robot.Native.get_robot_state()}");
                Console.WriteLine($"Estop reason: {robot.Native.get_estop_reason()}");

                if (runMotion)
                {
                    robot.StartSys();
                    robot.MoveJ(new double[] { 0.0, 0.0, 0.0, 0.0, 0.0, 0.5 }, 1.0, 0.5, 0.0, 0.0);
                    robot.MoveJ(new double[] { 0.0, 0.0, 0.0, 0.0, 0.0, 0.0 }, 1.0, 0.5, 0.0, 0.0);
                    robot.WaitMove();
                }
                else
                {
                    Console.WriteLine("Motion skipped. Pass --move to run the motion demo.");
                }
            }

            return 0;
        }
    }
}
